### PR TITLE
DIG-1463: Verify existence of genomic files

### PR DESCRIPTION
**Pull request by daisieh**
Added endpoint to verify existence of the files linked in a GenomicDrsObject. This method attempts to open the genomic file and get the samples listed; if it can do so, it compares the samples to the ones listed in the GenomicDrsObject. It returns `{"result": True}` if this all matches up and `{"result": False, "message": error message}` if not. The test in here is not very robust because htsget itself doesn't deal with samples very much: this is tested for real in test_integration.

Bonus: I have never bothered to add checksums and file sizes to DrsObjects as I should, based on the specification. Now that we've broken indexing out into a background process, I've hooked into that method to calculate checksums and sizes. This is now also tested in test_index_variantfile (which is run in test_integration as well).

### CODE REVIEW

File: htsget_server/drs_operations.py [hunk 1]
```diff
@@ -207,10 +216,15 @@ def _get_file_path(drs_file_obj_id):
...
             # we need to go to the access endpoint to get the url and file
-            (url, status_code) = _get_access_url(method["access_id"])
+            (url_obj, status_code) = _get_access_url(method["access_id"])
             result["status_code"] = status_code
             if status_code < 300:
-                result["path"] = url["url"]
+                result["path"] = url_obj["url"]
+                result["checksum"] = {
+                    "type": "etag",
+                    "checksum": url_obj["metadata"].etag
```
Comment (OrdiNeu) [inline htsget_server/drs_operations.py:225]: Where is this etag getting generated from? Is this related to http etags?

**Review by OrdiNeu — APPROVED** [review]
LGTM :heavy_check_mark:

File: htsget_server/drs_operations.py (hunk 1 above)
Comment (daisieh) [reply htsget_server/drs_operations.py:225]: This is an s3 checksum; it is part of the metadata in the Object class from the minio client SDK.
